packages/sanity/src/core/store/_legacy/document/document-pair/validation.ts

```diff
@@ -105,7 +106,7 @@ export const validation = memoize(
         // so only pass on documents if _other_ attributes changes
         return shallowEquals(omit(prev, '_rev', '_updatedAt'), omit(next, '_rev', '_updatedAt'))
       }),
-      share(),
+      shareLatestWithRefCount(),
```

Comment:
This is the line that fixes the issue. Since we were using share, re-subscribing to this observable didn't publish the current edit state of the document, leaving the validation in a pending state